test/e2e/bound_sa_token_test.go

```diff
@@ -59,34 +55,38 @@ func TestBoundTokenSignerController(t *testing.T) {
 
 	// The operand secret should be recreated after deletion.
 	t.Run("operand-secret-deletion", func(t *testing.T) {
-		test.WaitForKubeAPIServerClusterOperatorAvailableNotProgressingNotDegraded(t, configClient)
 
 		err := kubeClient.Secrets(targetNamespace).Delete(context.TODO(), tokenctl.SigningKeySecretName, metav1.DeleteOptions{})
 		require.NoError(t, err)
 		checkBoundTokenOperandSecret(t, kubeClient, regularTimeout, operatorSecret.Data)
 	})
 
 	// The operand config map should be recreated after deletion.
+	// Note: it will roll out a new version
 	t.Run("configmap-deletion", func(t *testing.T) {
-		test.WaitForKubeAPIServerClusterOperatorAvailableNotProgressingNotDegraded(t, configClient)
-
 		err := kubeClient.ConfigMaps(targetNamespace).Delete(context.TODO(), tokenctl.PublicKeyConfigMapName, metav1.DeleteOptions{})
 		require.NoError(t, err)
 		checkCertConfigMap(t, kubeClient, map[string]string{
 			"service-account-001.pub": string(operatorPublicKey),
 		})
+
+		// deletion triggers a roll-out - wait until a new version has been rolled out
+		test.WaitForAPIServerToBeOnTheSameRevision(t, kubeClient.Pods(operatorclient.TargetNamespace))
```

Comment:
don't we have to wait first that the operator has started progressing?

Reply:
nope, see - https://github.com/openshift/cluster-kube-apiserver-operator/pull/870#discussion_r435882090